packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts

```diff
@@ -146,6 +146,11 @@ const baseCases = [
 
 ruleTester.run('prefer-optional-chain', rule, {
   valid: [
+    'foo || {};',
+    'foo || ({} as any);',
+    '(foo || {})?.bar;', // This might seem stupid, but I'm not sure if we want to show suggestion for it
```

Comment:
I think this is a smart thing to add actually 😄. Checking for unnecessary optional chains is part of `no-unnecessary-condition`'s responsibility. 

```suggestion
    '(foo || {})?.bar;',
```